.checkpatch.conf

```diff
@@ -22,6 +22,7 @@
 --ignore MULTISTATEMENT_MACRO_USE_DO_WHILE
 --exclude ext
 --exclude samples/matter/.*/src
+--exclude applications/matter_weather_station/src
```

Comment:
how can you get such an exclude :)

Reply:
That is because Matter samples and now the application are written in C++, not in C. Checkpatch script is not able to correctly interpret .cpp files, so it fails every time returning false positive errors. When we were adding our samples to NCS some time ago, we agreed excluding our directory from checkpatch with NCS mainteners, so I guess approach should not differ in case of this particular application.